src/hooks/useActiveDescendantListFocus/__tests__/useActiveDescendantListFocus.jest.js

```diff
@@ -5,32 +5,109 @@ import userEvent from "@testing-library/user-event";
-      isHorizontalList: isHorizontal
-    })
-  );
+  const props = {
+    focusedElementRef: {
+      current: element
+    },
+    keepOptionSelected,
+    itemsIds: ITEM_IDS,
+    isItemSelectable: isItemSelectable,
+    onItemClick,
+    isHorizontalList: isHorizontal
+  };
+  return renderHook(argprops => useActiveDescendantListFocus({ ...props, ...argprops }));
 }
 
-function runListUnitTest(isHorizontal) {
+function runListUnitTest({ isHorizontal, keepOptionSelected }) {
   const moveForwardKey = isHorizontal ? "{arrowRight}" : "{arrowDown}";
   const oppositeMoveForwardKey = !isHorizontal ? "{arrowRight}" : "{arrowDown}";
+  const naturalIndex = keepOptionSelected ? 0 : undefined;
+  const naturalIndexAfterInitial = keepOptionSelected ? 0 : -1;
+
+  it("should focus same item after item changed", async () => {
+    const onItemClick = jest.fn();
+    const { result, rerender } = renderHookForTest({ onItemClick, isHorizontal, keepOptionSelected });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+
+    const beforeIndex = result.current.visualFocusItemIndex;
+    const beforeId = result.current.visualFocusItemId;
+
+    const removeItems = 2;
+    const itemsAfterRemove = ITEM_IDS.slice(removeItems);
+    rerender({ itemsIds: itemsAfterRemove });
+
+    if (!keepOptionSelected) {
+      expect(result.current.visualFocusItemId).toEqual(itemsAfterRemove[beforeIndex]);
+      expect(result.current.visualFocusItemIndex).toEqual(beforeIndex);
+    } else {
+      expect(result.current.visualFocusItemId).toEqual(beforeId);
+      expect(result.current.visualFocusItemIndex).toEqual(beforeIndex - removeItems);
+    }
+  });
+
+  it("should focus correct item on focus", async () => {
+    const onItemClick = jest.fn();
+    const { result } = renderHookForTest({ onItemClick, isHorizontal, keepOptionSelected });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+
+    expect(result.current.visualFocusItemIndex).toEqual(naturalIndex);
+  });
+
+  it("should focus correct item keyboard forward", async () => {
+    const onItemClick = jest.fn();
+    const { result } = renderHookForTest({ onItemClick, isHorizontal, keepOptionSelected });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+
+    const keyboardTimes = 2;
+    for (let i = 0; i < keyboardTimes; i++) {
+      act(() => {
+        // move visual focus to first item
+        userEvent.keyboard(moveForwardKey);
+      });
+    }
+
+    expect(result.current.visualFocusItemIndex).toEqual(naturalIndexAfterInitial + keyboardTimes);
```

Comment:
no need to split two tests here. just keep in variable the value of result.current.visualFocusItemIndex before pressing keyboard forward in variable + steps count and compare it to the result.current.visualFocusItemIndex after press